brave/src/main/java/com/linecorp/armeria/client/brave/BraveClient.java

```diff
@@ -78,16 +80,29 @@
     /**
      * Creates a new tracing {@link HttpClient} decorator using the specified {@link HttpTracing} instance.
      */
-    public static Function<? super HttpClient, BraveClient> newDecorator(
-            HttpTracing httpTracing) {
+    public static Function<? super HttpClient, BraveClient> newDecorator(HttpTracing httpTracing) {
         try {
             ensureScopeUsesRequestContext(httpTracing.tracing());
         } catch (IllegalStateException e) {
             logger.warn("{} - it is appropriate to ignore this warning if this client is not being used " +
                         "inside an Armeria server (e.g., this is a normal spring-mvc tomcat server).",
                         e.getMessage());
         }
-        return delegate -> new BraveClient(delegate, httpTracing);
+        final HttpRequestParser requestParser = httpTracing.clientRequestParser();
+        final HttpResponseParser responseParser = httpTracing.clientResponseParser();
+        if (requestParser != HttpRequestParser.DEFAULT && responseParser != HttpResponseParser.DEFAULT) {
+            return delegate -> new BraveClient(delegate, httpTracing);
+        }
+
+        // Override the brave default parsers to Armeria default parsers.
```

Comment:
for example, if someone is using vertx and they now switch to armeria, it will break how they parsed before unless they intentionally don't use the singletons. For example, the only way I can tell we can get back to "default" is to wrap the brave parsers again so that they aren't the same instance.